battle.go

```diff
@@ -143,6 +141,67 @@ func (b *Battle) derefActivePokemon(ap activePokemon) *Pokemon {
 	return (*b.Parties[b.agentParties[ap.AgentIdx]]).Pokemon[ap.PokemonIdx]
 }
 
+// A type that is necessary in order to implement the `Interface` interface, which is used by the sort package to sort.
+type apTurnOrder struct {
+	battle *Battle
+	active []activePokemon
+	turns  map[int]Turn
```

Comment:
We can use [slice sorting](https://yourbasic.org/golang/how-to-sort-in-go/#sort-with-custom-comparator) to avoid creating a new struct here, since we are sorting a slice of structs.

Reply:
Not with the current way the data is laid out. The order of active pokemon is relied upon.